packages/api-server/src/ws/methods.ts

```diff
@@ -34,15 +37,38 @@ export function wrapper(ws: any, _req: any) {
           method: "eth_subscription",
           params: {
             result: block,
-            subscription: "0x" + id.toString(16),
+            subscription: id,
           },
         };
         ws.send(JSON.stringify(obj));
       });
     });
   };
 
+  const logsListener = (logs: Log[]) => {
+    logsIds.forEach(async (id) => {
+      const query = logsQueryMaps.get(id);
+      if (!query) return;
+
+      const _result = await filterLogsByAddress(logs, query.address);
+      const result = await filterLogsByTopics(_result, query.topics || []);
+
+      if (result.length === 0) return;
+
+      const obj = {
+        jsonrpc: "2.0",
+        method: "eth_subscription",
+        params: {
+          result: result.map((log) => toApiLog(log)),
+          subscription: id,
+        },
+      };
+      ws.send(JSON.stringify(obj));
+    });
+  };
+
   blockEmitter.getEmitter().on("newHeads", blockListener);
+  blockEmitter.getEmitter().on("logs", logsListener);
```

Comment:
Release this source when close connection. See line 75 as an example.

Reply:
done